cmd/bosun/expr/expr.go

```diff
@@ -38,6 +38,14 @@ type State struct {
 	// LogstashElastic
 	logstashQueries      []elastic.SearchService
 	logstashElasticHosts []string
+
+	History AlertStatusProvider
+}
+
+// Alert Status Provider is used to provide information about alert results.
+// This facilitates alerts referencing other alerts, even when they go unknown or unevaluated.
+type AlertStatusProvider interface {
+	GetUnknownAndUnevaluatedAlertKeys(alertName string) (unknown, unevaluated []AlertKey)
```

Comment:
What do you think about inverting this API? Instead of this function, have two others: `IsUnknown(AlertKey)` and `IsUnevaluated(AlertKey)`. I think this would make the usage in conf.go simpler.